src/main/java/com/vmware/vipclient/i18n/base/DataSourceEnum.java

```diff
@@ -34,7 +43,13 @@ public MessageOpt createMessageOpt(MessagesDTO dto) {
 		public LocaleOpt createLocaleOpt() {
 			return new RemoteLocaleOpt();
 		}
+
+        @Override
+        public ComponentOpt createComponentOpt(BaseDTO dto) {
+            return new RemoteComponentOpt(dto);
+        }
     };
     public abstract MessageOpt createMessageOpt(MessagesDTO dto);
     public abstract LocaleOpt createLocaleOpt();
+    public abstract ComponentOpt createComponentOpt(BaseDTO dto);
```

Comment:
Need to add an interface to get supported locales. supported locales have nothing to do with displayLanguage. They are from names of bundle files.

Reply:
Why do you say "supported locales have nothing to do with displayLanguage"? The locale is composed of language tag and display name. The display name can be translated in multiple languages. The client is not using the display name now, but it can be used later on. 

Moreover, on the service side, we do not need 2 separate endpoints:
1. `GET /i18n/api/v2/translation/products/{productName}/versions/{version}/localelist`
2.`GET /i18n/api/v2/locale/supportedLanguageList`

Reply:
@Xiaochao8 , may I close this now, or do you not agree?

Reply:
In level 3, we only need the names from bundle file names. Please forget display names. It's unnecessary.

Reply:
I think you do not get my point. I KNOW that we are not using display names here. I said it up there in my comment.

The point is that we can USE THE SAME METHOD that L2 uses to get the list of supported locales. That method requires display language, yes -- which is no problem here because we can use the fallback locale as display language as well. Then just ignore the display names. Do you get it now? No need to write new code. Avoid redundant code like the redundant endpoints mentioned above in the service API.

Reply:
Also, why are you commenting on this line number? There is no displayLanguage here. What do you want to put in DataSourceEnum?

Reply:
Since you do not want to see a parameter in the method, I have created a parameterless LocaleService.getSupportedLanguages.